shared/actions/wallets.js

```diff
@@ -561,6 +568,31 @@ const maybeClearErrors = (state: TypedState) => {
   }
 }
 
+const maybeClearNewTxs = (action: RouteTreeGen.SwitchToPayload, state: TypedState) => {
+  const list = I.List(action.payload.path)
+  const root = list.first()
+  const settingsPath = getPath(state.routeTree.routeState, [Tabs.settingsTab])
+  const walletsPath = getPath(state.routeTree.routeState, [Tabs.walletsTab])
+  // If we're leaving from the Wallets tab, and the Wallets tab route
+  // was the main transaction list for an account, clear new txs.
+  // FIXME: The hardcoded routes here are fragile if routes change.
+  if (
```

Comment:
we'll make all this a lot simpler in the future , but for now i think it might be clearer to encode it more like (untested)

```
// outside this function, maybe in Constants?
const rootWalletTab = isMobile ? Tabs.settingsTab : Tabs.walletsTab
const isLookingAtWallet = routeState => getPath(routeState, rootWalletTab).get(isMobile ? 2 : 1) === 'wallet'

// then this turns into
if (rootTab !== Constants.rootWalletTab && Constants.isLookingAtWallet(lastRoute)) {

}
```

also, with the existing logic does the walletsPath.get(1) === 'wallet' work? isn't this the current path and not the previous one?

Reply:
@chrisnojima Constants change looks good.

Yeah, it works because this is on switchTab, so the path itself isn't changing, just which tab is selected.  We don't record the previous path anywhere -- the reducer runs before the saga and blows it away -- so looking at the previous path isn't easy right now.

If we'd like to expire the newPayments when you nav inside the wallets tab (e.g. go to transactions details and back, or click on a different account in the wallets list), as well as when you change tabs, I think the best way to get there might be to just add the expiry to those two click handlers, rather than hook into every navigateAppend/navigateTo and try to tell what the previous path was somehow.  What do you think?